File: Chapter5/BeakJoon/main.py
def dfs(x, y):
    if x <= -1 or x >= n or y <= -1 or y >= m:
        return False
    if graph[x][y] == 0:
        graph[x][y] = 1
        # 상하좌우 위치들도 재귀적으로 호출
        dfs(x-1, y)
        dfs(x, y-1)
        dfs(x+1, y)
        dfs(x, y+1)
        return False
    return False

n, m = map(int, input().split())

graph = []
for i in range(n):
    for j in range(m):
            # 현재 위치에서 dfs 수행
            if dfs(i, j) == True:
                result += 1

print(result)

File: Chapter5/BeakJoon/main.c
#include <stdio.h>


// 각 node가 연결된 정보를 표현 (2차원 list)
int graph[9][3] = {{}, {2,3,8}, {1,7}, {1,4,5}, {3,5}, {3,4}, {7}, {2,6,8}, {1,7}};
_Bool visitied;

void dfs(int x){
    visitied[x] = true;
    printf("%d", x);
    for(int i = 0; i < graph[x];i++){
        if(!visitied[i]){
            dfs(i);
        }
    }
}


void main(){
    dfs(1);
}